### DISCUSSION TITLE
Is there a way to prevent Renovate from searching certain repositories when scanning for latest dependencies

### DISCUSSION THREAD

**Discussion by jcchilton** [renovatebot/renovate#26621] [Request Help] [answered]
### How are you running Renovate?

None

### If you're self-hosting Renovate, tell us what version of Renovate you run.

Renovate Docker image: 37.127.0 

### If you're self-hosting Renovate, select which platform you are using.

None

### What is your question?

Is there a way to restrict Renovate from accessing certain repositories when it scans for latest version of dependencies? 

Example of what I mean:  

All of our projects require that dependencies be pulled from repositories in our private Artifactory.  However, there are some projects that also declare `MavenCentral` as a repository (In Gradle files).   For those projects that are referencing public repositories, is there a way to have Renovate not include those when looking for latest?  

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment]
You can use packageRules + registryUrls to manually specify the registries to search

**Reply by jcchilton** [reply]
Would the urls in `registryUrls` only be called if they have been declared by the Gradle files?  
If the urls in `registryUrls` contained urls for `artifactory/repo1, atrifactory/repo2`  but the project only requires `artifactory/repo1` we wouldn't want Renovate to also search in `artifactory/repo2`

**Reply by rarkins** [reply]
No, it's a simple override

**Reply by rarkins** [reply, marked as answer]
You can try disabling the maven hostname using hostRules.enabled = false if you want to block it altogether

**Reply by jcchilton** [reply]
I will give that try to see if will meet our needs

**Reply by jcchilton** [reply]
Thanks, using `hostName`  is workable.  
However, it would be preferable to have a whitelist (not the `registryUrls` override) of repositories  -- is that possible?

**Reply by rarkins** [reply]
You can configure packageRules using repository names or patterns (see https://docs.renovatebot.com/configuration-options/#matchrepositories) but not hostRules. Please describe in more detail what you mean by "whitelist of repositories" (to do or not do what?) so I can assess

**Reply by jcchilton** [reply]
By "whitelist of repositories" I mean:  A way to declare a list of repositories that are okay to use ( a superset instead of an overide).  That way  Renovate would use a project's `gradle,build` file and would only look in the repositories from the gradle file as long as they are also in the "whitelist".

**Reply by rarkins** [reply]
Not possible

**Reply by jcchilton** [reply]
Thanks!
